### PR TITLE
Text transform prop pill

### PR DESCRIPTION
**Pull request by batistaDev1113**
#### Screens
formPillReact
formPillRails

PillReact

PillRails

#### Breaking Changes

[Yes/No (Explain)]

Text transform prop have been added to both Form Pill kit and Pill kit

#### Runway Ticket URL

https://nitro.powerhrg.com/runway/backlog_items/PLAY-40

#### How to test this

Pass the textTransform prop to the Form Pill && Pill kits(React) and text_transform to Form Pill & Pill kit (Rails) respectively.

#### Checklist:

- [ ] **LABELS** Add a label: `enhancement`, `bug`, `improvement`, `new kit`, `deprecated`, or `breaking`. See [Changelog & Labels](https://github.com/powerhome/playbook/wiki/Changelog-&-Labels) for details.
- [ ] **DEPLOY** Please add the `Milano` label when you are ready for a review.
- [ ] **SCREENSHOT** Please add a screen shot or two.
- [ ] **SPECS** Please cover your changes with specs.
- [ ] **READ DOCS** Please make sure you have read and understand the [Playbook Release Process](https://github.com/powerhome/playbook/wiki/Playbook-Releases)

### CODE REVIEW

File: playbook/app/pb_kits/playbook/pb_pill/docs/_pill_variants.html.erb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+            <%= pb_rails("pill", props: { text: "info" , variant: "info" }) %>
+
+
+
+                <%= pb_rails("pill", props: { text: "neutral" , variant: "neutral" }) %>
+
+
+
+                    <%= pb_rails("pill", props: { text: "primary" , variant: "primary" }) %>
```
Comment (jasoncypret) [inline playbook/app/pb_kits/playbook/pb_pill/docs/_pill_variants.html.erb:21]: Can you correct this indentation on this file?
```suggestion
<%= pb_rails("pill", props: { text: "success" , variant: "success" }) %>
<%= pb_rails("pill", props: { text: "error" , variant: "error" }) %>
<%= pb_rails("pill", props: { text: "warning" , variant: "warning" }) %>
<%= pb_rails("pill", props: { text: "info" , variant: "info" }) %>
<%= pb_rails("pill", props: { text: "neutral" , variant: "neutral" }) %>
<%= pb_rails("pill", props: { text: "primary" , variant: "primary" }) %>
```

File: playbook/app/pb_kits/playbook/pb_pill/docs/_pill_variants.jsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import React from 'react'
+import Pill from '../_pill.jsx'
+
+const PillVariants = () => {
+  return (
+    <div>
+      <Pill
+          text="success"
+          variant="success"
+      />
```
Comment (jasoncypret) [inline playbook/app/pb_kits/playbook/pb_pill/docs/_pill_variants.jsx:10]: I believe you need to pass global props to the kit so the dark mode is retained. Click the dark mode button to check this. Also look at another react example 

```suggestion
const PillVariants = (props) => {
  return (
    <div>
      <Pill
          text="success"
          variant="success"
          {...props}
      />
```

File: playbook/app/pb_kits/playbook/pb_pill/docs/_pill_default.jsx [hunk 3]
```diff
@@ -5,54 +5,6 @@ const PillDefault = () => {
   return (
```
Comment (jasoncypret) [inline playbook/app/pb_kits/playbook/pb_pill/docs/_pill_default.jsx:5]: This needs the {...props} added as well

File: playbook/app/pb_kits/playbook/pb_pill/docs/_pill_example.jsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+const PillExample = () => {
+  return (
+    <div>
+      <Pill
+          text="Success"
+          textTransform="none"
+          variant="success"
+      />
+    </div>
```
Comment (jasoncypret) [inline playbook/app/pb_kits/playbook/pb_pill/docs/_pill_example.jsx:12]: {...props} added

```suggestion
const PillExample = (props) => {
  return (
    <div>
      <Pill
          text="Success"
          textTransform="none"
          variant="success"
          {...props}
      />
    </div>
```

**Review by jasoncypret — APPROVED** [review]

**Comment by garettarrowood** [comment]
Great! It's nice to have a workaround to get Rails pill kits to be case sensitive.

Why default this to `downcase` though? Just so there will not be surprises in Nitro? Because since the new convention for pills is to use capital letters, I think they would all be welcome surprises.

TLDR; Can we change the default to "none"?

**Comment by jasoncypret** [comment]
@garettarrowood that is for backward compatibility. It is currently lowercase by default. This is just retaining that so no views are broken. At some future point, we can change the default, but we will need to check & change every view in Nitro that uses this kit.

**Comment by garettarrowood** [comment]
@jasoncypret - I see your point. You are 100% right. -- I think in regards to capitalization though, the greatest amount of breakage would only be displaying a pill like "ACTIVE" or "ActiVE" instead of "active". And in many cases, pills will just become "Active" instead of "active" because the dev passed it a capitalize string already.

Regardless - thank you for fixing/amending this kit! Very appreciated!

And you make a good point. It might be good to do a pass through pills in Nitro and just fix them all. Very easy and safe upgrade.

**Comment by garettarrowood** [comment]
... something that could be allocated to an Apprentice maybe :)
